virttest/qemu_vm.py

```diff
@@ -3231,14 +3229,20 @@ def get_vnc_port(self):
 
         return self.vnc_port
 
-    def get_vcpu_pids(self, vcpu_thread_pattern):
+    def get_vcpu_pids(self, debug=True):
         """
         Return the list of vcpu PIDs
 
         :return: the list of vcpu PIDs
         """
-        return [int(_) for _ in re.findall(vcpu_thread_pattern,
-                                           str(self.monitor.info("cpus")))]
+        vcpus_info = self.monitor.info_cpus(debug=debug)
+        if isinstance(self.monitor, qemu_monitor.QMPMonitor):
+            vcpu_pids = [int(vcpu_info.get('thread-id', vcpu_info.get('thread_id')))
```

Comment:
Typo?  output:
```
[{"arch": "x86", "thread-id": 5957, "props": {"core-id": 0, "thread-id": 0, "die-id": 0, "socket-id": 0}, "qom-path": "/machine/unattached/device[0]", "cpu-index": 0, "target": "x86_64"}]
```

Reply:
The key value returned by `query-cpus` is `thread_id`, and returned by `query-cpus-fast` is `thread-id`.

Reply:
They both are thread-id in my test. 
```
{ "execute": "query-cpus-fast"}
{"return": [{"arch": "x86", "thread-id": 13216, "props": {"core-id": 0, "thread-id": 0, "socket-id": 0}, "qom-path": "/machine/unattached/device[0]", "cpu-index": 0, "target": "x86_64"}, {"arch": "x86", "thread-id": 13218, "props": {"core-id": 0, "thread-id": 0, "socket-id": 1}, "qom-path": "/machine/unattached/device[2]", "cpu-index": 1, "target": "x86_64"}, {"arch": "x86", "thread-id": 13219, "props": {"core-id": 0, "thread-id": 0, "socket-id": 2}, "qom-path": "/machine/unattached/device[3]", "cpu-index": 2, "target": "x86_64"}, {"arch": "x86", "thread-id": 13220, "props": {"core-id": 0, "thread-id": 0, "socket-id": 3}, "qom-path": "/machine/unattached/device[4]", "cpu-index": 3, "target": "x86_64"}]}

{ "execute": "query-cpus"}
{"return": [{"arch": "x86", "current": true, "props": {"core-id": 0, "thread-id": 0, "socket-id": 0}, "CPU": 0, "qom_path": "/machine/unattached/device[0]", "pc": 1029620, "halted": true, "thread_id": 13216}, {"arch": "x86", "current": false, "props": {"core-id": 0, "thread-id": 0, "socket-id": 1}, "CPU": 1, "qom_path": "/machine/unattached/device[2]", "pc": 1036412, "halted": true, "thread_id": 13218}, {"arch": "x86", "current": false, "props": {"core-id": 0, "thread-id": 0, "socket-id": 2}, "CPU": 2, "qom_path": "/machine/unattached/device[3]", "pc": 1036412, "halted": true, "thread_id": 13219}, {"arch": "x86", "current": false, "props": {"core-id": 0, "thread-id": 0, "socket-id": 3}, "CPU": 3, "qom_path": "/machine/unattached/device[4]", "pc": 1036412, "halted": true, "thread_id": 13220}]}
```

Reply:
@PaulYuuu  got your point, your are right.
```
{"execute":"query-cpus-fast"}                    
{"return": [{"arch": "x86", "**thread-id**": 5957, "props": {"core-id": 0, "thread-id": 0, "die-id": 0, "socket-id": 0}, "qom-path": "/machine/unattached/device[0]", "cpu-index": 0, "target": "x86_64"}]}
{"execute":"query-cpus"}
{"return": [{"arch": "x86", "current": true, "props": {"core-id": 0, "thread-id": 0, "die-id": 0, "socket-id": 0}, "CPU": 0, "qom_path": "/machine/unattached/device[0]", "pc": 1030118, "halted": true, "**thread_id**": 5957}]}

```